### PR TITLE
Sync default-ingress-cert to the console namespace

### PR DESCRIPTION
**Pull request by jhadvig**
Revering and renaming the revert of the https://github.com/openshift/console-operator/pull/328 based on https://github.com/openshift/enhancements/pull/126

/assign @benjaminapetersen

### CODE REVIEW

File: hack/test-unit.sh [hunk 1]
```diff
@@ -10,6 +10,7 @@ PACKAGES_TO_TEST=(
     "github.com/openshift/console-operator/pkg/console/operator"
     "github.com/openshift/console-operator/pkg/console/starter"
     "github.com/openshift/console-operator/pkg/console/subresource/configmap"
+    "github.com/openshift/console-operator/pkg/console/subresource/consoleserver"
```
Comment (benjaminapetersen) [inline hack/test-unit.sh]: This is a good one to bring back.
Tempting to separate it out into a separate PR, just in case backports of this PR come into play.

File: pkg/console/operator/sync_v400.go [hunk 2]
```diff
@@ -82,6 +82,9 @@ func (co *consoleOperator) sync_v400(updatedOperatorConfig *operatorv1.Console,
 	// The sync loop may not settle, we are unable to honor it in current state.
 	status.HandleProgressingOrDegraded(updatedOperatorConfig, "CustomLogoSync", customLogoErrReason, customLogoError)
 
+	defaultIngressCertConfigMap, defaultIngressCertErrReason, defaultIngressCertError := co.ValidateDefaultIngressCertConfigMap()
+	status.HandleProgressingOrDegraded(updatedOperatorConfig, "DefaultIngressCertValidation", defaultIngressCertErrReason, defaultIngressCertError)
```
Comment (benjaminapetersen) [inline pkg/console/operator/sync_v400.go]: One thing about bringing this back,  we need to make sure we don't panic again, per this bug:

https://bugzilla.redhat.com/show_bug.cgi?id=1778930

File: pkg/console/operator/sync_v400.go [hunk 3]
```diff
@@ -408,6 +423,20 @@ func (c *consoleOperator) SyncCustomLogoConfigMap(operatorConfig *operatorsv1.Co
 	return okToMount, reason, err
 }
 
+func (c *consoleOperator) ValidateDefaultIngressCertConfigMap() (defaultIngressCert *corev1.ConfigMap, reason string, err error) {
+	defaultIngressCertConfigMap, err := c.configMapClient.ConfigMaps(api.OpenShiftConsoleNamespace).Get(api.DefaultIngressCertConfigMapName, metav1.GetOptions{})
```
Comment (benjaminapetersen) [inline pkg/console/operator/sync_v400.go]: I bet right in here is where we just have to double check about that `panic` bug.

**Comment by benjaminapetersen** [comment]
I believe we need this in `4.4`,`4.3`,`4.2`,`4.1`.  Backporting with 11 file changes may be fun :) Gonna verify that.

File: pkg/console/operator/sync_v400.go (hunk 2 above)
Comment (jhadvig) [reply pkg/console/operator/sync_v400.go]: true, we are missing error check.. need to add:

```go
if defaultIngressCertError != nil {
    return defaultIngressCertError
}
```

File: hack/test-unit.sh (hunk 1 above)
Comment (jhadvig) [reply hack/test-unit.sh]: Do you mean just to have the `config_builder_test.go` and `config_merger_test.go` disabled? cause we will need to update the `config_builder.go` with the necessary logic anyway

**Comment by jhadvig** [comment]
@benjaminapetersen PR updated.
Regarding the splitting of the PR, it doesn't make sense since we will have to do it manually for 4.0 and 4.1 completely, since we are missing the `subresources` dir. For 4.2 we need to change just two files manually.

File: pkg/console/subresource/configmap/configmap_test.go [hunk 4]
```diff
@@ -108,6 +156,7 @@ customization:
 						Host: host,
 					},
 				},
+				useDefaultCAFile: true,
```
Comment (benjaminapetersen) [inline pkg/console/subresource/configmap/configmap_test.go]: We have at least one that verifies this generates correctly with `useDefaultCAFile: true`?

File: pkg/console/subresource/configmap/configmap_test.go [hunk 5]
```diff
@@ -80,6 +82,52 @@ servingInfo:
...
+					Namespace:   api.OpenShiftConsoleNamespace,
+					Labels:      map[string]string{"app": api.OpenShiftConsoleName},
+					Annotations: map[string]string{},
+				},
+				Data: map[string]string{configKey: `kind: ConsoleConfig
+apiVersion: console.openshift.io/v1
+auth:
+  clientID: console
+  clientSecretFile: /var/oauth-config/clientSecret
+  oauthEndpointCAFile: /var/default-ingrass-cert/ca-bundle.crt
```
Comment (benjaminapetersen) [inline pkg/console/subresource/configmap/configmap_test.go]: Oop, `ingrass` -> `ingress`.

File: pkg/console/subresource/consoleserver/config_builder.go [hunk 6]
```diff
@@ -9,8 +9,9 @@ import (
 )
 
 const (
-	clientSecretFilePath    = "/var/oauth-config/clientSecret"
-	oauthEndpointCAFilePath = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+	clientSecretFilePath       = "/var/oauth-config/clientSecret"
+	defaultIngressCertFilePath = "/var/default-ingrass-cert/ca-bundle.crt"
```
Comment (benjaminapetersen) [inline pkg/console/subresource/consoleserver/config_builder.go]: Oop, `ingrass` -> `ingress`.

File: pkg/console/subresource/deployment/deployment.go [hunk 7]
```diff
@@ -419,6 +422,12 @@ func defaultVolumeConfig() []volumeConfig {
 			path:        "/var/service-ca",
 			isConfigMap: true,
 		},
+		{
+			name:        api.DefaultIngressCertConfigMapName,
+			readOnly:    true,
+			path:        "/var/default-ingrass-cert",
```
Comment (benjaminapetersen) [inline pkg/console/subresource/deployment/deployment.go]: `ingrass` -> `ingress`.

File: pkg/console/subresource/deployment/deployment_test.go [hunk 8]
```diff
@@ -506,6 +535,11 @@ func Test_consoleVolumeMounts(t *testing.T) {
 					ReadOnly:  true,
 					MountPath: "/var/service-ca",
 				},
+				{
+					Name:      api.DefaultIngressCertConfigMapName,
+					ReadOnly:  true,
+					MountPath: "/var/default-ingrass-cert",
```
Comment (benjaminapetersen) [inline pkg/console/subresource/deployment/deployment_test.go]: `ingrass` -> `ingress`.
I think once these are updated we are good here.

**Comment by benjaminapetersen** [comment]
/lgtm 

thanks!
